py3status/modules/keyboard_locks.py

```diff
@@ -62,28 +55,16 @@ def keyboard_lock(self):
             scrlock_color = self.py3.COLOR_GOOD
             scrlock_icon = self.icon_scrlock_on
 
+        caps = self.py3.composite_create({'full_text': capslock_icon, 'color': capslock_color})
+        num = self.py3.composite_create({'full_text': numlock_icon, 'color': numlock_color})
+        scr = self.py3.composite_create({'full_text': scrlock_icon, 'color': scrlock_color})
+
+        full_text = self.py3.safe_format(
+            self.format, {'caps': caps}, {'num': num}, {'scr': scr})
```

Comment:
`{'caps': caps, 'num': num, 'scr': scr}`

should fix it.  We can pass a composite just like any other variable to be substituted